.pre-commit-config.yaml

```diff
@@ -1,4 +1,18 @@
 repos:
+  # See https://pre-commit.com/hooks.html for info on hooks
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.3.0
+    hooks:
+      - id: check-added-large-files
+      - id: check-ast
+      - id: check-case-conflict
+      - id: check-toml
+      - id: check-yaml
+      - id: debug-statements
+      - id: detect-private-key
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
```

Comment:
Do all of the files currently pass all these checks? 

I suspect I've failed somewhere 🤣

Reply:
My understanding is that Pre-commit (rightfully) only checks changed lines. Otherwise, I would not have even been able to submit this pull request in its current form, because (among other violations) Pre-commit would have flagged all the Flake8 issues that you and I have both already submitted other PRs to address. 😅 

So I believe the idea is for these changes to apply to lines changed in future commits. That said, I think the only violations I see when running across the entire code base that relate to these new hooks are:
* Trim Trailing Whitespace
* Fix End of Files

… most of which (if not all) should theoretically already be addressed in other pending PRs.

_Helpful note:_ You can manually run all Pre-commit hooks on all local files via:

	pre-commit run --all-files --show-diff-on-failure

Reply:
Thanks for this, I think it is a nice addition. 

As you say there are a few files which fail the whitespace and end of file checks. Could you add a cleanup for those here? (I tried to push a fix to your branch but it wouldn't let me push).

Reply:
Weird — I thought I checked the box that allows maintainers to make changes. In any case, I just pushed a commit that is the result of the above command, applying whitespace & line ending fixes.